src/common/storage/project/cache.js

```diff
@@ -95,102 +120,98 @@ define(['common/util/assert', 'common/storage/constants'], function (ASSERT, CON
             callback(null, obj);
         };
 
+        /**
+         * Loads the necessary objects for the nodes corresponding to paths and inserts them in the cache.
+         * If the rootKey is empty or does not exist - it won't attempt to load any nodes.
+         * @param {string} rootKey
+         * @param {string[]} paths
+         * @param {function(err)} callback
+         */
         this.loadPaths = function (rootKey, paths, callback) {
-            ASSERT(typeof rootKey === 'string' && paths instanceof Array && typeof callback === 'function');
             logger.debug('loadPaths', {metadata: {rootKey: rootKey, paths: paths}});
 
-            function getFromCache(hash) {
-                var obj = cache[hash],
-                    commitId;
-
-                if (typeof obj === 'undefined') {
-                    obj = backup[hash];
-
-                    if (typeof obj === 'undefined') {
-                        for (commitId in self.queuedPersists) {
-                            if (self.queuedPersists.hasOwnProperty(commitId) && self.queuedPersists[commitId][key]) {
-                                obj = self.queuedPersists[commitId][key];
-                                break;
-                            }
-                        }
-                    }
-                }
-
-                return obj;
-            }
-
-            var objects = {},
+            var cachedObjects = {},
                 excludes = [],
+                rootObj = getFromCache(rootKey),
+                i = paths.length,
+                j,
                 pathArray,
                 fullyCovered,
                 obj,
-                rootObj = getFromCache(rootKey),
-                key, i = paths.length,
-                j;
-
-            if (typeof rootObj !== 'undefined') {
-                excludes.push(rootKey);
-                objects[rootKey] = rootObj;
-
-                while (i--) {
-                    fullyCovered = true;
-                    pathArray = paths[i].split('/');
-                    if(pathArray.length > 1){
-                        pathArray.shift();
-                    }
-                    obj = rootObj;
-                    for (j = 0; j < pathArray.length; j += 1) {
-                        key = obj[pathArray[j]];
-                        if (key) {
-                            obj = getFromCache(key);
-                            if (typeof obj !== 'undefined') {
-                                excludes.push(key);
-                                objects[key] = obj;
-                            } else {
-                                fullyCovered = false;
-                                break;
-                            }
+
+                key;
+
+            if (!rootKey || !rootObj) {
+                logger.debug('rootKey empty or rootObject was not loaded in cache, rootKey:', rootKey);
+                callback(null);
+                return;
+            }
```

Comment:
Yes, I will return on faulty rootKey though (without it the server could not figure out where to start).